src/crystal/system/unix/getrandom.cr

```diff
@@ -3,6 +3,8 @@
 require "c/unistd"
 require "syscall"
 
+private Syscall.def_syscall getrandom, LibC::SSizeT, buf : UInt8*, buflen : LibC::SizeT, flags : UInt32
```

Comment:
I think the method should still be namespaced to `Crystal::System::Random`, not in the top-level namespace.

Reply:
This will conflict with `Crystal::System::Random.getrandom`, which already exists. One possible solution would be to make `def_syscall` always generate a `sys_` prefix and then rename the existing `sys_getrandom` to `sys_getrandom_wrap`. What do you think?

Reply:
Depends on how we want to proceed with syscalls in the stdlib API. 
If we end up collecting all the methods in `Syscall` as you initially proposed, there's not much reason to think about naming conflicts. So let's wait and see.

I think it's best to leave like that for now.

Reply:
Why not bind to Libc?

Reply:
It would require glibc 2.25. Is that OK?

Reply:
Not sure. It is something super new?

Reply:
That said, it's fine for now

Reply:
@asterite This particular change is just cleanup changing an existing syscall that is already used in the current code to instead use a little more structured way to define syscalls. I don't know if this particular one exist in LibC yet.

However, the relevant syscalls driving the feature is the ones related to io_uring. Those are from Linux 5.3 or something like that, and LibC has not included them yet. There is a more user friendly C library for uring that the kernel maintainers also maintain at https://github.com/axboe/liburing , but that can also not be expected to exist as a distro-packaged library.

Reply:
I agree with @straight-shoota , it feels a bit weird that the syscall is defined here at the top-level. Privately, but at the top-level.

Why not all syscalls are defined directly inside the `Syscall` namespace? Then you can do `Syscall.getrandom` from anywhere, and it's very clear that `getrandom` is a syscall...

Reply:
But there's a point in having syscalls within the `Crystal::System` namespace, to make sure they're not consider part of the public API. So I think it would be best to go with `Crystal::System::Syscall`.